steps/actions/cache.dhall

```diff
@@ -1,17 +1,23 @@
 let Step = ../../schemas/Step.dhall
 
-in  λ(args : { path : Text, key : Text, hashFile : Text }) →
-      Step::{
-      , id = None Text
-      , name = Some "${args.path} cache"
-      , uses = Some "actions/cache@v2"
-      , run = None Text
-      , `with` = Some
-          ( toMap
-              { path = args.path
-              , key =
-                  "\${{ runner.os }}-${args.key}-\${{ hashFiles('${args.hashFile}') }}"
-              , restore-keys = args.key
-              }
-          )
-      }
+let Text/concatMapSep =
+      https://prelude.dhall-lang.org/v17.1.0/Text/concatMapSep.dhall sha256:c272aca80a607bc5963d1fcb38819e7e0d3e72ac4d02b1183b1afb6a91340840
+
+in  λ(args : { path : Text, key : Text, hashFiles : List Text }) →
+      let hashFilesArg =
+            Text/concatMapSep "," Text (λ(x : Text) → "'${x}'") args.hashFiles
+
+      in  Step::{
+          , id = None Text
+          , name = Some "${args.path} cache"
+          , uses = Some "actions/cache@v2"
+          , run = None Text
+          , with = Some
```

Comment:
```suggestion
          , `with` = Some
```

Reply:
Dhall 1.34.0's formatter and its linter both try to remove the backticks. Do you want them forced in?

Reply:
Hm, wasn't aware of that... but looking at the release notes I can't see that  ... `with` is still a reserved keyword so escaping should still be necessary 🤷